src/sentry/static/sentry/app/utils/queryString.tsx

```diff
@@ -35,6 +35,11 @@ export function appendTagCondition(
   key: string,
   value: string
 ): string {
+  // Use the explicit tag syntax to handle the case when the tag name (key) is a
+  // reserved keyword.
+  // https://docs.sentry.io/workflow/search/?platform=node#explicit-tag-syntax
+  key = `tags[${key}]`;
```

Comment:
Could we only wrap in `tags[]` for known problematic keys?

Reply:
I agree. Wondering if we can only wrap on reserved tags.

Reply:
I agree. Do we make up a new list of reserved keys? That would go out of sync with the server but is possibly better than doing nothing.

Reply:
I believe we could have an ENUM with all the reserved keys and use it to check if the key is reserved or not ... If you want, I can help you with it. 

Btw, where can we find the list of reserved keys?

Reply:
We have a list of fields in `app/views/eventsV2/eventQueryParams.tsx.` The `FIELDS` constant is the list of reserved fields that are used in discover, and partly in issues. Issue search adds [additional properties](https://github.com/getsentry/sentry/blob/35fefafff0a05fed4a4f80a0fbeee4fa18538f21/src/sentry/api/issue_search.py#L25-L40) and I'm not sure we have a list of those in the clientside code.

Reply:
@markstory  thank you for pointing this out ;-)